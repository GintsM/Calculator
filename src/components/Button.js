import PropTypes from 'prop-types';

const Button = ({ className, value, onEvent }) => (
  <button type="button" className={className} onClick={() => onEvent(value)}>
    {value}
  </button>
);

Button.propTypes = {
  className: PropTypes.string,
  value: PropTypes.string,
  onEvent: PropTypes.func.isRequired,
};

Button.defaultProps = {
  className: '',
  value: '',
};

export default Button;
