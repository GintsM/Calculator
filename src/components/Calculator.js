import React from 'react';
import Calc from '../logic/calculate';

class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      total: null,
      next: null,
      operation: null,
    };
  }

  render() {
    const handler = (buttonName) => {
      const name = buttonName.target.innerText;
      const { total, next, operation } = Calc(this.state, name);
      this.setState({ total, next, operation });
    };
    const { total, next, operation } = this.state;

    const display = () => {
      if (next) {
        return next;
      }

      if (operation) {
        return operation;
      }
      if (total) {
        return total;
      }
      return 0;
    };

    return (
      <div className="wrapper">
        <input className="input-field" value={display()} type="text" maxLength="0" onChange={() => ''} />
        <div className="btn-wrapper">
          <button className="btn" type="button" onClick={handler}>AC</button>
          <button className="btn" type="button" onClick={handler}>+/-</button>
          <button className="btn" type="button" onClick={handler}>%</button>
          <button className="btn light" type="button" onClick={handler}>รท</button>
        </div>
        <div className="btn-wrapper">
          <button className="btn" type="button" onClick={handler}>7</button>
          <button className="btn" type="button" onClick={handler}>8</button>
          <button className="btn" type="button" onClick={handler}>9</button>
          <button className="btn light" type="button" onClick={handler}>x</button>
        </div>
        <div className="btn-wrapper">
          <button className="btn" type="button" onClick={handler}>4</button>
          <button className="btn" type="button" onClick={handler}>5</button>
          <button className="btn" type="button" onClick={handler}>6</button>
          <button className="btn light" type="button" onClick={handler}>-</button>

        </div>
        <div className="btn-wrapper">
          <button className="btn" type="button" onClick={handler}>1</button>
          <button className="btn" type="button" onClick={handler}>2</button>
          <button className="btn" type="button" onClick={handler}>3</button>
          <button className="btn light" type="button" onClick={handler}>+</button>
        </div>
        <div className="btn-wrapper">
          <button className="btn" type="button" onClick={handler}>0</button>
          <button className="btn light-end" type="button" onClick={handler}>=</button>
        </div>
      </div>
    );
  }
}

export default Calculator;
