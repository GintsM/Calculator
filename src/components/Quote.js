const Quote = () => (
  <div>
    <article>
      <h3>
        All mathematicians share a sense of amazement
        over the infinite deh3th and mysterious beauty and usefulness of mathematics.
      </h3>
      <p>
        - MARTIN GARDNER
      </p>
    </article>
  </div>
);

export default Quote;
