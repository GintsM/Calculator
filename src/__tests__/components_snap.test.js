import { cleanup } from '@testing-library/react';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Button from '../components/Button';
import Home from '../components/Home';
import Quote from '../components/Quote';
import Calculator from '../components/Calculator';
import Nav from '../components/navbar';

afterEach(() => cleanup());

describe('matches components', () => {
  test('matches snapshot button', () => {
    const tree = renderer.create(<Button value="AC" className="btn" onEvent={() => ''} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('matches snapshot Home', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('matches snapshot Quote', () => {
    const tree = renderer.create(<Quote />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('matches snapshot Calculator', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('matches snapshot Navbar', () => {
    const navtree = renderer.create(
      <BrowserRouter>
        <Nav />
        <Routes>
          <Route exact path="/" element={<Home />} />
          <Route path="/calculator" element={<Calculator />} />
          <Route path="/quotes" element={<Quote />} />
        </Routes>
      </BrowserRouter>
    ).toJSON();
    expect(navtree).toMatchSnapshot();
  });
});
