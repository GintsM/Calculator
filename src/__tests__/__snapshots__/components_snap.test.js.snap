// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches components matches snapshot Calculator 1`] = `
<div
  className="same"
>
  <h2>
    Let's do some math!
  </h2>
  <div
    className="wrapper"
  >
    <input
      className="input-field"
      data-testid="hello"
      maxLength="0"
      readOnly={true}
      type="text"
      value={0}
    />
    <div
      className="btn-wrapper"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="btn light"
        onClick={[Function]}
        type="button"
      >
        ÷
      </button>
    </div>
    <div
      className="btn-wrapper"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="btn light"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </div>
    <div
      className="btn-wrapper"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="btn light"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="btn-wrapper"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="btn light"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="btn-wrapper"
    >
      <button
        className="btn wide"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="btn light"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`matches components matches snapshot Home 1`] = `
<div
  className="home"
>
  <article>
    <h2>
      Welcome to our page
    </h2>
    <p>
      This project is built using REACT, and will be Calculator for simple mathematical operations will add more description, with building the project.
    </p>
    <p>
      If you aren’t satisfied with the build tool and configuration choices, you can \`eject\` at any time. This command will remove the single build dependency from your project.
    </p>
    <p>
      Instead, it will copy all the configuration files and the transitive dependencies (webpack, Babel, ESLint, etc) right into your project so you have full control over them. All of the commands except \`eject\` will still work, but they will point to the copied scripts so you can tweak them. At this point you’re on your own.
    </p>
    <p>
      You don’t have to ever use \`eject\`. The curated feature set is suitable for small and middle deployments, and you shouldn’t feel obligated to use this feature. However we understand that this tool wouldn’t be useful if you couldn’t customize it when you are ready for it.
    </p>
  </article>
</div>
`;

exports[`matches components matches snapshot Navbar 1`] = `
Array [
  <nav
    className="same"
  >
    <h1
      className="heading"
    >
      Math Magicians
    </h1>
    <ul>
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
      <a
        href="/quotes"
        onClick={[Function]}
      >
        Quote
      </a>
    </ul>
  </nav>,
  <div
    className="home"
  >
    <article>
      <h2>
        Welcome to our page
      </h2>
      <p>
        This project is built using REACT, and will be Calculator for simple mathematical operations will add more description, with building the project.
      </p>
      <p>
        If you aren’t satisfied with the build tool and configuration choices, you can \`eject\` at any time. This command will remove the single build dependency from your project.
      </p>
      <p>
        Instead, it will copy all the configuration files and the transitive dependencies (webpack, Babel, ESLint, etc) right into your project so you have full control over them. All of the commands except \`eject\` will still work, but they will point to the copied scripts so you can tweak them. At this point you’re on your own.
      </p>
      <p>
        You don’t have to ever use \`eject\`. The curated feature set is suitable for small and middle deployments, and you shouldn’t feel obligated to use this feature. However we understand that this tool wouldn’t be useful if you couldn’t customize it when you are ready for it.
      </p>
    </article>
  </div>,
]
`;

exports[`matches components matches snapshot Quote 1`] = `
<div>
  <article>
    <h3>
      All mathematicians share a sense of amazement over the infinite deh3th and mysterious beauty and usefulness of mathematics.
    </h3>
    <p>
      - MARTIN GARDNER
    </p>
  </article>
</div>
`;

exports[`matches components matches snapshot button 1`] = `
<button
  className="btn"
  onClick={[Function]}
  type="button"
>
  AC
</button>
`;
